internal/sidecar/service/networkfence.go

```diff
@@ -115,3 +115,48 @@ func getCIDRS(cidr []string) []*fence.CIDR {
 	}
 	return cidrs
 }
+
+// GetFenceClients fetches required information from kubernetes cluster and calls
+// CSI-Addons GetFenceClients service.
+func (ns *NetworkFenceServer) GetFenceClients(
+	ctx context.Context,
+	req *proto.FenceClientsRequest) (*proto.FenceClientsResponse, error) {
+	data := map[string]string{}
+	var err error
+	// Get the secrets from the k8s cluster if secret name or namespace is provided
+	if req.GetSecretName() != "" || req.GetSecretNamespace() != "" {
+		data, err = kube.GetSecret(ctx, ns.kubeClient, req.GetSecretName(), req.GetSecretNamespace())
+		if err != nil {
+			klog.Errorf("Failed to get secret %s in namespace %s: %v", req.GetSecretName(), req.GetSecretNamespace(), err)
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		}
+	}
+
+	getFenceClientRequest := fence.GetFenceClientsRequest{
+		Parameters: req.GetParameters(),
+		Secrets:    data,
+	}
+
+	resp, err := ns.controllerClient.GetFenceClients(ctx, &getFenceClientRequest)
+	if err != nil {
+		klog.Errorf("Failed to get fence clients: %v", err)
+		return nil, err
+	}
+	response := &proto.FenceClientsResponse{}
+	for _, client := range resp.Clients {
+		response.Clients = append(response.Clients, &proto.ClientDetails{
+			Id:        client.Id,
+			Addresses: convertCIDRToAddress(client.Addresses),
+		})
+	}
+
+	return response, nil
+}
+
+func convertCIDRToAddress(cidr []*fence.CIDR) []string {
```

Comment:
A CIDR is a notation for a range of addresses, not a single address. Either the name of the function is wrong, or it does the wrong thing.

Reply:
The function name was wrong, reworded it